crates/mapdiffbot2/src/job_processor.rs

```diff
@@ -102,136 +72,98 @@ fn render(
         "hide-space,hide-invisible,random",
     );
 
+    trace!("Initializing vars");
+
     // ADDED MAPS
-    let added_directory = format!("{}/a", output_dir.display());
+    let added_directory = format!("{}/a", out_dir.display());
     let added_directory = Path::new(&added_directory);
     let added_errors = Default::default();
 
     // MODIFIED MAPS
-    let modified_directory = format!("{}/m", output_dir.display());
+    let modified_directory = format!("{}/m", out_dir.display());
     let modified_directory = Path::new(&modified_directory);
     let modified_before_errors = Default::default();
     let modified_after_errors = Default::default();
 
-    let removed_directory = format!("{}/r", output_dir.display());
+    let removed_directory = format!("{}/r", out_dir.display());
     let removed_directory = Path::new(&removed_directory);
     let removed_errors = Default::default();
 
-    let base_maps =
-        with_repo_dir(&base.repo, || load_maps(modified_files)).context("Loading base maps")?;
-    let head_maps = with_checkout(&base.repo, &pull_branch, || load_maps(modified_files))
+    trace!("Loading maps");
+
+    let base_maps = with_checkout_and_dir(&base_branch, repo, &path, || load_maps(modified_files))
+        .context("Loading base maps")?;
+    let head_maps = with_checkout_and_dir(&head_branch, repo, &path, || load_maps(modified_files))
         .context("Loading head maps")?;
+
+    trace!("Getting bounding boxes");
+
     let modified_maps = get_map_diff_bounding_boxes(base_maps, head_maps);
 
-    //let now = Instant::now();
+    trace!("Rendering removed/modified maps");
     // You might think to yourself, wtf is going on here?
     // And you'd be right.
-    let removed_maps = with_repo_dir(&base.repo, || {
-        let results = rayon::join(
-            || {
-                render_map_regions(
-                    &base_context,
-                    &modified_maps.befores,
-                    &head_render_passes,
-                    modified_directory,
-                    "before.png",
-                    &modified_before_errors,
-                )
-                .context("Rendering modified before maps")
-            },
-            || {
-                let maps = load_maps_with_whole_map_regions(removed_files)
-                    .context("Loading removed maps")?;
-
-                render_map_regions(
-                    &base_context,
-                    &maps,
-                    &base_render_passes,
-                    removed_directory,
-                    "removed.png",
-                    &removed_errors,
-                )
-                .context("Rendering removed maps")?;
-
-                Ok(maps)
-            },
-        );
-        results.0?;
-        results.1
-        //eprintln!("Base maps took {}ms", now.elapsed().as_millis());
+    let removed_maps = with_checkout_and_dir(&base_branch, repo, &path, || {
+        render_map_regions(
```

Comment:
This is no longer multithreaded? Is that okay?

Reply:
yes